src/kv/RocksDBStore.cc

```diff
@@ -577,10 +577,10 @@ void RocksDBStore::RocksDBTransactionImpl::set(
 	     rocksdb::Slice(to_set_bl.buffers().front().c_str(),
 			    to_set_bl.length()));
   } else {
-    // make a copy
-    bufferlist val = to_set_bl;
-    bat.Put(rocksdb::Slice(key),
-	     rocksdb::Slice(val.c_str(), val.length()));
+    rocksdb::Slice key_slice(key);
+    rocksdb::Slice value_slices[to_set_bl.buffers().size()];
+    bat.Put(nullptr, rocksdb::SliceParts(&key_slice, 1),
+            rocksdb::SlicePartsi(&value_slices, to_set_bl.buffers().size()));
```

Comment:
I don't see the part where value_slices is populated with the bufferptr data pointers...